test/commands/test/lib/included-files-parser-test.ts

```diff
@@ -1,7 +1,112 @@
 import { IFileDescriptionJson } from "../../../../src/commands/test/lib/test-manifest-reader";
 import { expect } from "chai";
-import { parseIncludedFiles } from "../../../../src/commands/test/lib/included-files-parser";
+import { copyIncludedFiles, parseIncludedFiles } from "../../../../src/commands/test/lib/included-files-parser";
+import * as xmlUtil from "../../../../src/commands/test/lib/xml-util";
+import * as pfs from "../../../../src/util/misc/promisfied-fs";
 import * as os from "os";
+import * as Sinon from "sinon";
+
+describe("copyIncludedFiles", () => {
+  let sandbox: Sinon.SinonSandbox = null;
+
+  beforeEach(() => {
+    sandbox = Sinon.sandbox.create();
+  });
+  afterEach(() => {
+      sandbox.restore();
+  });
+
+  it("should do nothing if null include is sent", async () => {
+    const inputManifest = { schemaVersion: "1.0.0", cliVersion: "1.0.0",  applicationFile: "",
+    files: ["file1", "file2"], testFramework: { name: "", data: "" } };
+    const expectedManifest = { schemaVersion: "1.0.0", cliVersion: "1.0.0",  applicationFile: "",
```

Comment:
I have returned a manifest to make testing easier - however I have left it also updating the referenced manifest (I don't want to risk introducing regressions this late in the PR process).

I had to monkey about with the assignment of `inputManifest` to enable it to get the references object (setting the value in the before function).

Reply:
I think it should be one or the other:

1) return a modified manifest and do not modify the existing one.
2) leave it as it is.

I would actually prefer 2) over the current

"I don't want to risk introducing regressions this late in the PR process)."
The tests are there for us to detect a regression, hence be comfortable with make a change like the one I suggested.

With option 1) there would be no need for monkeying :-)

Reply:
OK, I've moved it back to 2) purely because there are other places where we send the manifest as a reference - I agree that it would be better to clone and return the manifest but I don't want to have inconsistencies in how different areas behave - if I had more time I'd refactor all these areas so they work by returning a new copy of the manifest.

However, the refactoring does mean that I no longer need to reference the manifest at all to test my changes (the filter step is a distinct step as discussed).

Reply:
> The tests are there for us to detect a regression, hence be comfortable with make a change like the one I suggested.

Worth noting that not all of this is my code and the project as a whole is light on tests so I'm naturally scared of making too many structural changes. I would _love_ to spend some time improving the test coverage and testability of the CLI though!